.travis.yml

```diff
@@ -37,19 +36,28 @@ _manylinux-build: &manylinux-build
           branch: master
           repo: pypa/manylinux
 
-jobs:
-  include:
-    - stage: "Patch glibc"
+_patch-glibc: &patch-glibc
       env:
         - PLATFORM="x86_64"
       before_install:
-        # Load cached docker images
-        - if [[ -d $HOME/docker ]]; then ls $HOME/docker/*.tar.gz | xargs -I {file} sh -c "zcat {file} | docker load"; fi
-      script:
-        - PLATFORM=$PLATFORM TRAVIS_COMMIT=$TRAVIS_COMMIT ./build.sh glibc_only
+        - docker/load_cached_images.sh
       before_cache:
-        # Save tagged docker images
-        - mkdir -p $HOME/docker && docker images -a --filter='dangling=false' --format '{{.Repository}}:{{.Tag}} {{.ID}}' | grep 'centos-with-vsyscall:latest' | xargs -n 2 -t sh -c 'test -e $HOME/docker/$1.tar.gz || docker save $0 | gzip -2 > $HOME/docker/$1.tar.gz'
+        - docker/cache_images.sh
+
+jobs:
+  include:
+    - <<: *patch-glibc
+      stage: "Build patched x86_64 glibc for x86_64 image"
+      script:
+        - PLATFORM=$PLATFORM docker/glibc/build.sh 32
+    - <<: *patch-glibc
+      stage: "Build patched i686 glibc for x86_64 image"
+      script:
+        - PLATFORM=$PLATFORM docker/glibc/build.sh 64
```

Comment:
What happens if `stage` is the same ? Would it get built in parallel thus speeding-up the build ? Does it interfere with caching ?

Reply:
Unfortunately, parallel build would indeed interfere with caching here. The problem is that even if you build a single target inside a Dockerfile, everything preceding it gets built, not just its dependencies. This makes 64-bit glibc effectively depend on the 32-bit one.